xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/NodeSubscription.java

```diff
@@ -829,7 +829,7 @@ void sendLastPublishedItem(PublishedItem publishedItem) {
      * @return true if the specified user is allowed to modify or cancel the subscription.
      */
     boolean canModify(JID user) {
-        return user.equals(getJID()) || user.equals(getOwner()) || node.getService().isServiceAdmin(user);
+        return user.equals(getJID()) || user.toBareJID().equals(getOwner().toBareJID()) || node.getService().isServiceAdmin(user);
```

Comment:
Good as far as it goes. Question; should there also be a `user.toBareJID().equals(getJID().toBareJID())` as well? It's quite possible that the modifying JID has a different resource from the subscribing JID, but be the same user.

Reply:
Agreed.

Reply:
On second thought: no, maybe not. The 'JID' is the target for notifications to be sent to. Although this would nearly always be equal to the owner, it might not be. If it's not, then I'm not sure if giving all resources of the receiving entity should be allowed access.

I'm thinking that this is largely a moot point, as Openfire will mostly decide based on 'owner' anyway. I'll leave the JID-handling as-is.